File: evidence/generated-systems/golang/healthcare-on-golang/healthcare-on-golang/internal/model/LabResult.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// LabResult Declaration
//==============================================================
type LabResult struct {
    gorm.Model
     ResultCode                                    string
    IssuedDate                                                            time.Time
    LaboratoryOrderId         *uint
    LaboratoryOrder           *LaboratoryOrder `gorm:"foreignKey:LaboratoryOrderId"`
     Observations           []Observation `gorm:"foreignKey:ObservationsFromLabResultId"`
    LaboratoryId         *uint
    Laboratory           *Laboratory `gorm:"foreignKey:LaboratoryId"`
    Status                      ResultStatus

// parent associations as their child

}

File: evidence/generated-systems/django/healthcare-on-django/healthcareOnDjango/models/LabResult.py
from django.db import models
from healthcareOnDjango.models.ResultStatus import ResultStatus

#======================================================================
# 
# Encapsulates data for model LabResult
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class LabResult Declaration
#======================================================================
class LabResult (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	resultCode = models.CharField(max_length=200, null=True)
	issuedDate = models.CharField(max_length=64, null=True)
	laboratoryOrder = models.ForeignKey('LaboratoryOrder', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	observations = models.ManyToManyField('Observation',  blank=True, related_name='+')
	laboratory = models.ForeignKey('Laboratory', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in ResultStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.resultCode
		str = str + self.issuedDate
		str = str + self.status
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "LabResult";
    
	def objectType(self):
		return "LabResult";
